rust/kernel/error.rs

```diff
@@ -61,6 +63,26 @@ impl Error {
     }
 }
 
+impl fmt::Debug for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

Comment:
I like the idea that we could have names instead of values for the `Debug::fmt` of `Error`. However, I think it's better that we don't need to call `errname` if we already have the corresponding const, e.g. `Error::EINVAL`, because having the const actually means we have names for the error numbers at Rust side.

Thoughts?

Reply:
We don't currently have constants for all errors defined in Rust. Plus, defining them in Rust essentially requires us to have a giant match, or define a lookup table like `errname.c` did. It might be better just to reuse `errname`, C side needs it anyway.